src/server/generic_family.cc

```diff
@@ -623,17 +625,12 @@ void OpScan(const OpArgs& op_args, const ScanOpts& scan_opts, uint64_t* cursor,
   // 10k Traverses
   const size_t limit = 10000;
   do {
+    cur = prime_table->Traverse(
+        cur, [&](PrimeIterator it) { cnt += ScanCb(op_args, it, scan_opts, vec); });
+    ++buckets_iterated;
     if (buckets_iterated >= limit) {
-      buckets_iterated = 0;
-      util::ThisFiber::Yield();
-    }
-    {
-      // Disable flush journal changes to prevent preemtion in traverse.
-      journal::JournalFlushGuard journal_flush_guard(op_args.shard->journal());
-      cur = prime_table->Traverse(
-          cur, [&](PrimeIterator it) { cnt += ScanCb(op_args, it, scan_opts, vec); });
+      break;
```

Comment:
This should be fine. We can still over spin a little bit, because the called of `OpScan` does not return unless:
```
 677     // Break after kMaxScanTimeMs.                                                                  
 678     uint64_t time_now_ms = GetCurrentTimeMs();                                                      
 679     if (time_now_ms > db_cntx.time_now_ms + kMaxScanTimeMs) {                                       
 680       break;                                                                                        
 681     } 
 ```
 
 or when the limit is reached. Basically, our worse latency is now the timeout (as opposed to `OpScan` potentially iterate the whole db table which would be unbounded time depending on the shard size)